tests/integration/golang/helpers/client.go

```diff
@@ -33,117 +56,133 @@ func NewAimApiClient(baseURL string) *HttpClient {
+	if c.request != nil {
+		data, err := json.Marshal(c.request)
+		if err != nil {
+			return eris.Wrap(err, "error marshaling request object")
+		}
+		requestBody = bytes.NewBuffer(data)
 	}
 
-	body, err := io.ReadAll(resp.Body)
-	defer resp.Body.Close()
+	// 2. build actual URL.
+	u, err := url.Parse(fmt.Sprintf("%s%s%s", c.baseURL, c.basePath, fmt.Sprintf(uri, values...)))
 	if err != nil {
-		return nil, eris.Wrap(err, "error reading streaming response")
+		return eris.Wrap(err, "error building url")
+	}
+	// 3. if params were provided then add params to actual url.
+	if c.params != nil {
+		switch reflect.ValueOf(c.params).Kind() {
+		case reflect.Struct:
+			query, err := urlquery.Marshal(c.params)
+			if err != nil {
+				return eris.New("error marshaling params")
+			}
+			u.RawQuery = string(query)
+		case reflect.Map:
+			query := u.Query()
+			for key, value := range c.params.(map[any]any) {
+				query.Set(fmt.Sprintf("%v", key), fmt.Sprintf("%v", value))
+			}
+			u.RawQuery = query.Encode()
+		default:
+			return eris.New("unsupported type of params. should be struct or map[any]any")
+		}
 	}
 
-	return body, nil
-}
-
-// doRequest do request of any http method
-func (c HttpClient) doRequest(httpMethod string, uri string, response interface{}, requestBody io.Reader) error {
-	// 1. create actual request object.
+	// 4. create actual request object.
+	// if HttpMethod was not provided, then by default use HttpMethodGet.
 	req, err := http.NewRequestWithContext(
-		context.Background(),
-		httpMethod,
-		StrReplace(
-			fmt.Sprintf(
-				"%s%s%s",
-				c.baseURL,
-				c.basePath,
-				uri,
-			),
-			[]string{},
-			[]interface{}{},
-		),
-		requestBody,
+		context.Background(), string(c.method), u.String(), requestBody,
 	)
 	if err != nil {
 		return eris.Wrap(err, "error creating request")
 	}
-	req.Header.Set("Content-Type", "application/json")
 
-	// 3. send request data.
+	// 5. if headers were provided, then attach them.
+	// by default attach `"Content-Type", "application/json"`
+	if c.headers != nil {
+		for key, value := range c.headers {
+			req.Header.Set(key, value)
+		}
+	}
+
+	// 6. send request data.
 	resp, err := c.client.Do(req)
 	if err != nil {
 		return eris.Wrap(err, "error doing request")
 	}
 
-	// 4. read and check response data.
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return eris.Wrap(err, "error reading response data")
-	}
-	defer resp.Body.Close()
-	if err := json.Unmarshal(body, response); err != nil {
-		return eris.Wrap(err, "error unmarshaling response data")
+	// 7. read and check response data.
+	if c.response != nil {
+		switch c.responseType {
+		case ResponseTypeJSON:
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return eris.Wrap(err, "error reading response data")
+			}
+			defer resp.Body.Close()
+			if err := json.Unmarshal(body, c.response); err != nil {
+				return eris.Wrap(err, "error unmarshaling response data")
+			}
+		case ResponseTypeStream:
```

Comment:
I believe this `ResponseTypeBuffer`. We should also have an actual `ResponseTypeStream` that returns an `io.ReadCloser`

Reply:
so do I correctly understood you?:
- you are talking about to rename `ResponseTypeStream` -> `ResponseTypeBuffer`?
- you want to return `io.ReadCloser`? or? not sure that understood you fully?

Reply:
* yes, please rename `ResponseTypeStream` to `ResponseTypeBuffer`, as it does return an entire buffer
* I am suggesting to _also_ add an actual response type for streaming, called `ResponseTypeStream` and that returns an `io.ReadCloser` — this doesn't have to be done in this PR, but maybe add a TODO comment about it

Reply:
1. - fixed
2. - added TODO section.